### PR TITLE
Industry best practises

### PR DESCRIPTION
**Pull request by rfussien**
Hi,

I made some best practices improvements to your package.
Hope it'll be merged, and 1.0.1 will be released.

Regards,

### CODE REVIEW

**Comment by halilim** [comment]
Wow, a big change for a small repo :) This will take some time to actually review as I'm no longer a PHPer (would you like to take over as the maintainer, or be a collaborator?).

Technically I'm unsure about bumping version as there is no functionality change, even if I'd be more than happy to do it.

**Comment by rfussien** [comment]
Hi,
"I'm unsure about bumping version as there is no code/dependency/functionality change"
That's why I suggested to increase the Patch part of the version, which following the semver description is just fix with no functionality and dependency change.

I see no problem about being a contributor. However I'd appreciate if you could take some time to set up scrutinizer.

Warm regards,
Rémi

File: .travis.yml [hunk 1]
```diff
@@ -5,4 +5,32 @@ php:
...
+sudo: false
+
+## Cache composer
+cache:
+  directories:
+    - $HOME/.composer/cache
+
+matrix:
+  include:
+    - php: 7.0
```
Comment (rfussien) [inline .travis.yml]: This is not a duplicate. It takes the needed packages at their lowest accepted version.

File: benchmark.php [hunk 2]
```diff
@@ -43,12 +49,8 @@
...
 }
 
-
 $time = microtime(true) - $startTime;
 echo "\n";
 echo "method      : $method\n";
 echo "elem count  : $ct\n";
 echo "time passed : $time s\n";
-
-// Useless since the memory used by libxml libraries is not reported
```
Comment (rfussien) [inline benchmark.php]: I even think this file is not relevant and could be removed. 
And just for info, if you get back to php, there is an awesome benchmarking tool called PhpBench to do this kind of benchmark.

File: .gitignore [hunk 3]
```diff
@@ -1 +1,3 @@
-phpunit.xml
```
Comment (rfussien) [inline .gitignore]: yep, you're totally right, my bad.

**Pull request merged**